src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java

```diff
@@ -0,0 +1,673 @@
+            }
+
+            return getParticipantById(convert2Id);
+        }
+    }
+
+    @Override
+    public List<Participant> getAllParticipantsByType(ParticipantType participantType){
+        List<Participant> participants = participantService.getAllParticipantByType(participantType);
+        if(participants.isEmpty()){
+            throw new NotFound("No participants in type of: " + participantType.toString() + " found");
+        }
+
+        return participants;
+    }
+
+    @Override
+    public List<Participant> getAllParticipants(){
+        List<Participant> participants = participantService.getAllParticipant();
+        if(participants.isEmpty()){
+            throw new NotFound("No participant found");
+        }
+
+        return participants;
+    }
+
+    @Override
+    public void deleteParticipantById(Long id){
+        //Participant checkParticipant = participantService.getParticipantById(id);
+        if(!checkParticipantExistsById(id)){
+            throw new NotFound("Participant with id: " + id + " couldn't found");
+        }
+
+        participantService.deleteParticipantById(id);
+    }
+
+    @Override
+    public void deleteParticipantByUserName(String userName){
+        //Participant checkParticipant = participantService.getParticipantByUserName(userName);
+        if(!checkParticipantExistsByUserName(userName)){
+            throw new NotFound("Participant with username: " + userName + " couldn't found");
+        }
+
+        participantService.deleteParticipantByUserName(userName);
+    }
+
+    @Override
+    public void updateParticipant(Long id, Participant participant){
+        //Participant checkParticipant = participantService.getParticipantById(id);
+        if(!checkParticipantExistsById(id)){
+            throw new NotFound("Participant with id: " + id + " couldn't found");
+        }
+
+        if((participant.getId() != id) && (participant.getId() != null)){
+            throw new Conflict("Participant data has id: " + participant.getId() + " but update id: " + id);
+        }
+
+        participant.setId(id);
+        participantService.saveParticipant(participant);
+    }
+
+    @Override
+    public void saveParticipant(Participant participant){
+        participantService.saveParticipant(participant);
+    }
+
+    @Override
+    public void addComponent(Component component, String owner, Long parentComponentId){
+        if(owner.isEmpty() || owner == null){
+            throw new Conflict("Owner of a component can't be empty");
+        }
+
+        //Participant checkParticipant = participantService.getParticipantByUserName(owner);
+        if(!checkParticipantExistsByUserName(owner)){
+            throw new NotFound("Participant with username: " + owner + " couldn't found");
+        }
+
+        //Component checkComponent = icalService.getComponentById(parentComponentId);
+        if(!checkComponentExistsById(parentComponentId) && parentComponentId != null){
```

Comment:
second null check unnecessarily I guess, firstly checkComponentExistsById function will work, and if it is working, parentComponentId cannot be null.